src/machine/asm/execute_x64.S

```diff
@@ -225,9 +206,48 @@
   POSTCALL; \
 2:
 
+#define CHECK_READ_VERSION0(address_reg, length) \
+  movq address_reg, TEMP1; \
+  shr $CKB_VM_ASM_MEMORY_FRAME_SHIFTS, TEMP1; \
+  movq CKB_VM_ASM_ASM_CORE_MACHINE_OFFSET_LAST_READ_FRAME(MACHINE), TEMP2; \
+  cmp TEMP2, TEMP1; \
+  jne 3f; \
+  movq address_reg, TEMP2; \
+  addq $length, TEMP2; \
+  shr $CKB_VM_ASM_MEMORY_FRAME_SHIFTS, TEMP2; \
+  cmp TEMP2, TEMP1; \
+  je 2f; \
```

Comment:
Where is label 2? I only found label 3.

Reply:
Label 2 here refers to the label in `CHECK_READ_FRAMES`

Reply:
The way I look at this is: `CHECK_READ_FRAMES` is an internal macro used across `CHECK_READ_VERSION0` and `CHECK_READ_VERSION1`